test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +90,9 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if(inputCommand.equals("add")||inputCommand.equals("delete")||inputCommand.equals("clear")) {
```

Comment:
Since you already have the method isMutating, you can simply call this method to check whether it is a mutating command, instead of manually listing them out. Also, to be complete, you should have an `else` branch that confirms the file was not updated for non mutating commands.